components/style/properties/helpers/animated_properties.mako.rs

```diff
@@ -2383,11 +2562,132 @@ impl Animatable for TransformList {
     }
 }
 
+/// A helper function to retrieve the pixel length and percentage value.
+fn extract_pixel_calc_value(lop: &LengthOrPercentage) -> (f64, CSSFloat) {
+    match lop {
+        &LengthOrPercentage::Length(au) => (au.to_f64_px(), 0.),
+        &LengthOrPercentage::Percentage(percent) => (0., percent.0),
+        &LengthOrPercentage::Calc(calc) => (calc.length().to_f64_px(), calc.percentage())
+    }
+}
+
+/// Compute the squared distance of two transform lists.
+// This might not be the most useful definition of distance. It might be better, for example,
+// to trace the distance travelled by a point as its transform is interpolated between the two
+// lists. That, however, proves to be quite complicated so we take a simple approach for now.
+// See https://bugzilla.mozilla.org/show_bug.cgi?id=1318591#c0.
+fn compute_transform_lists_squared_distance(from_list: &[TransformOperation],
+                                            to_list: &[TransformOperation])
+                                            -> Result<SquaredDistance, ()> {
+    let mut squared_distance = SquaredDistance::Value(0.);
+    for (from, to) in from_list.iter().zip(to_list.iter()) {
```

Comment:
You don't need a `for` loop at all here, just `map` on the iterator and `sum` it at the end.